src/tests/ftest/launch.py

```diff
@@ -253,14 +253,17 @@ def set_test_environment(args):
             print("  {}: {}".format(key, os.environ[key]))
 
 
-def get_output(cmd, check=True):
+def get_output(cmd, check=True, raise_exception=False):
     """Get the output of given command executed on this host.
 
     Args:
         cmd (list): command from which to obtain the output
-        check (bool, optional): whether to raise an exception and exit the
-            program if the exit status of the command is non-zero. Defaults
-            to True.
+        check (bool, optional): whether to {emit an error and exit the
+            program|raise an exception} if the exit status of the command
+            is non-zero. Defaults to True.
+        raise_exception (bool, optional): whether to raise an exception
+            rather than exit the program if the exit status of the command
+            is non-zero. Defaults to False.
```

Comment:
What do you think about reusing `check` as more of a "what to do if there is a command error" (`error_action` ??) flag with three states:

- 0 = Ignore the error
- 1 = Report the error and exit the program
- 2 = Raise a RuntimeError exeception

So the code (currently lines 277-284) looks like this:
```
    if retcode:
        msg = "Error executing '{}':\n\tOutput:\n{}".format(
            " ".join(cmd),
        if error_action == 1:
            print(msg)
            exit(1)
        elif error_action == 2:
            raise RuntimeError(msg)
```

Reply:
Yeah, could do that.  Was just trying to avoid breaking the existing API.  But given that it's consumers are all in the same file, perhaps not a big deal.

Reply:
Alternatively - and probably how it should have been done in the first place:

```
def run_command(cmd):
    """Get the output of given command executed on this host.

    Args:
        cmd (list): command from which to obtain the output

    Raises:
        RuntimeError: if the command fails

    Returns:
        str: command output

    """
    print("Running {}".format(" ".join(cmd)))
    process = subprocess.Popen(
        cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
    stdout, _ = process.communicate()
    retcode = process.poll()
    if retcode:
        raise RuntimeError(
            "Error executing '{}':\n\tOutput:\n{}".format(
                " ".join(cmd), stdout))
    return stdout

def get_output(cmd, check=True):
    """Get the output of given command executed on this host.

    Args:
        cmd (list): command from which to obtain the output
        check (bool, optional): whether to {emit an error and exit the
            program|raise an exception} if the exit status of the command
            is non-zero. Defaults to True.

    Returns:
        str: command output
    """
    try:
        stdout = run_command(cmd)
    except RuntimeError as error:
        if check:
            print(error)
            exit(1)
    return stdout
```

Reply:
Yeah, I agree and like this the best.  Will implement.